trackintel/analysis/location_identification.py

```diff
@@ -0,0 +1,214 @@
+import numpy as np
+import pandas as pd
+
+
+def location_identifier(sps, pre_filter=True, recipe="FREQ"):
+    """Find home and work/location for each user with different recipes.
+
+    Parameters
+    ----------
+    sps : Geodataframe (as trackintel staypoints)
+        Staypoints with column "location_id".
+
+    pre_filter : bool, default True
+        Prefiltering the staypoints to exclude locations with not enough data.
+        The filter function can also be accessed via `pre_filter`.
+
+    recipe : {'FREQ'}, default "FREQ"
+        Choose which recipe to use.
+        - FREQ: Select the most visited location
+        - PASS: just for decoration.
+
+    Returns
+    -------
+    Geodataframe (as trackintel staypoints)
+        With additional column assigning one of three location identies {'home', 'work', None}.
+
+    Note
+    ----
+    The recipe are adapted from [1]. The original algorithms count the distinct hours at a
+    location as the home location is derived from geo-tagged tweets. We directly sum the time
+    spent at a location as our data model includes that.
+
+    References
+    ----------
+    Chen, Qingqing, and Ate Poorthuis. 2021.
+    ‘Identifying Home Locations in Human Mobility Data: An Open-Source R Package for Comparison and Reproducibility’.
+    International Journal of Geographical Information Science 0 (0): 1–24.
+    https://doi.org/10.1080/13658816.2021.1887489.
+
+    Examples
+    --------
+    >>> ti.analysis.location_identifier(sps, pre_filter=True, recipe="FREQ")
+    """
+    # what do we do here?
+    # we take the gdf and assert two things 1. is staypoint 2. has location_id column
+    assert sps.as_staypoints
+    if "location_id" not in sps.columns:
+        raise KeyError(
+            (
+                "To derive location activities the GeoDataFrame (as trackintel staypoints)must have a column "
+                f"named 'location_id' but it has [{', '.join(sps.columns)}]"
+            )
+        )
+    # then hand it to to the filter function if necessary.
+    if pre_filter:
+        f = pre_filter_locations()
+        sps = sps[f]
+    if recipe == "FREQ":
+        return freq_recipe(sps, "home", "work")
+    else:
+        raise ValueError(f"Recipe {recipe} does not exist.")
+
+
+def pre_filter_locations(
+    sps,
+    agg_level="user",
+    thresh_min_sp=10,
+    thresh_min_loc=10,
+    thresh_sp_at_loc=10,
+    thresh_loc_time=1,
+    thresh_loc_period=pd.Timedelta("5h"),
+):
```

Comment:
IMO, this `pre_filter` is quite specific to the `freq_recipe` function that we use, and sometimes too strict? would it be possible to **not** using some of the filters? for example, if I do not want a `thresh_min_sp` filter, what should I input here?

I realize this is the design in the original article and might be too minor for this first pull request

EDIT: I notice in the test that we can disable the filters by setting them to 0. Maybe state this in the docstring